From: Bernd Zeimetz <bernd@bzed.de>
Date: Sat, 19 Feb 2011 17:46:59 +0100
Subject: xgps/xgpsspeed: Check for a usable display before writing to it.

This avoids coredumps in case DISPLAY is not usable.
See http://bugs.debian.org/602925 for details.
---
 xgps      |    2 ++
 xgpsspeed |    2 ++
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/xgps b/xgps
index fe4ce7c..85a077d 100755
--- a/xgps
+++ b/xgps
@@ -313,6 +313,8 @@ class Base:
         self.ais_latch = False
   
         self.window = gtk.Window(gtk.WINDOW_TOPLEVEL)
+        if not self.window.get_display():
+            raise Exception("Can't open display")
         self.window.set_title("xgps")
         self.window.connect("delete_event", self.delete_event)
         self.window.set_resizable(False)
diff --git a/xgpsspeed b/xgpsspeed
index 7f12b8d..2db696f 100755
--- a/xgpsspeed
+++ b/xgpsspeed
@@ -252,6 +252,8 @@ class Main(object):
         self.debug = debug
         self.speed_unit = speed_unit
         self.window = gtk.Window(gtk.WINDOW_TOPLEVEL)
+        if not self.window.get_display():
+            raise Exception("Can't open display")
         self.window.set_title('xgpsspeed')
         self.widget = Speedometer(speed_unit=self.speed_unit)
         self.window.connect('delete_event', self.delete_event)
-- 
